swan-cern/files/swan_config_cern.py

```diff
@@ -39,34 +56,29 @@ def _init_eos_secret(self):
         except Exception as e:
             raise ValueError("Could not create required user credential")
 
-
         # ITHADOOP-819 - Ports need to be opened using service creation, and later assigning allocated service nodeport to a pod 
         # Create V1Secret with eos token
-        try:
-            secret_data = client.V1Secret()
-
-            secret_meta = client.V1ObjectMeta()
-            secret_meta.name = eos_secret_name
-            secret_meta.namespace = swan_container_namespace
-            secret_meta.labels = {
-                "swan_user": username
-            }
-            secret_data.metadata = secret_meta
-            secret_data.data = {}
-            secret_data.data['krb5cc'] = eos_token_base64
+        secret_data = V1Secret()
+
+        secret_meta = V1ObjectMeta()
+        secret_meta.name = eos_secret_name
+        secret_meta.namespace = swan_container_namespace
+        secret_meta.labels = {
+            "swan_user": username
+        }
+        secret_data.metadata = secret_meta
+        secret_data.data = {}
+        secret_data.data['krb5cc'] = eos_token_base64
 
+        try:
+            # eos-tokens secret is cleaned when user session ends, so try creating it
```

Comment:
Btw, this reminds me that we split the code... This means that we have similar logic in 2 different places (we assume here that it will be created, but it's done somewhere else... And in that place we assumed it was created here...). I'm ok with leaving it here for now, but it's not the ideal solution. Let's open a PR upstream to fix the async call and then we put them back together (unless you think they should all live in the spawner).

Reply:
Yes, creation and deletion are in different places now. I'd prefer they are all in the spawner, but that does not mean they can't be hooks. Regarding the contribution upstream, I added a comment to this task https://its.cern.ch/jira/browse/SWAN-85